src/Repository.ts

```diff
@@ -278,6 +280,31 @@ export class MongoRepository<DOC, DTO = DOC> {
     return document;
   }
 
+  /**
+   * Apply functions to a record based on the type of event
+   *
+   * @private
+   * @param {string} type any of the valid types, PRE_KEY POST_KEY
+   * @param {string[]} fns any of the valid functions: update, updateOne, save, create, find, findOne, findMany
+   * @param {*} newDocument The document to apply functions to
+   * @param {*} oldDocument The original document before changes were applied
+   * @returns {Promise<DOC>}
+   * @memberof MongoRepository
+   */
+  protected async invokeEvents(type: string, fns: string[], newDocument: any, oldDocument?: any): Promise<any> {
+    for (const fn of fns) {
+      const events = Reflect.getMetadata(`${type}_${fn}`, this) || [];
+      for (const event of events) {
+        newDocument = event.bind(this)(newDocument, oldDocument, fn);
```

Comment:
think it might be better (for the future) to start wrapping these additional args into an object instead. (little strange that oldDocument is optional, but fn isn't _really_, as it's always provided)
```
interface RepoEventArgs: {
originalDocument?: DOC;
eventType: PRE_KEY | POST_KEY;
eventOp: 'update' | 'find' | ...
}
eventArgs: RepoEventArgs = { originalDocument: oldDocument, eventType: type, eventOp: fn };
event.bind(this)(newDocument, eventArgs);
```

Reply:
yeah, sure

Reply:
updated